cli/flox-rust-sdk/src/providers/build.rs

```diff
@@ -1076,16 +1075,74 @@ mod tests {
                 mkdir $out
                 echo -n "$FOO" > $out/{file_name}
             """
-        "#};
+            sandbox = "{}"
+        "#, if sandbox { "pure" } else { "off" }};
 
         let (flox, _temp_dir_handle) = flox_instance();
         let mut env = new_path_environment(&flox, &manifest);
         let env_path = env.parent_path().unwrap();
 
+        if sandbox {
+            let _git = GitCommandProvider::init(&env_path, false).unwrap();
+        }
+
         assert_build_status(&flox, &mut env, &package_name, None, true);
         assert_build_file(&env_path, &package_name, &file_name, &file_content);
     }
 
+    #[test]
+    fn build_uses_var_from_manifest_sandbox_off() {
+        build_uses_var_from_manifest(false);
+    }
+
+    #[test]
+    fn build_uses_var_from_manifest_pure() {
+        build_uses_var_from_manifest(true);
+    }
+
+    #[test]
+    fn vars_not_set_at_runtime() {
+        let package_name = String::from("foo");
+        let file_path = String::from("bin/print_var");
+        let file_content = String::from("some content");
+        let var = "FOO";
+
+        let manifest = formatdoc! {r#"
+            version = 1
+
+            [vars]
+            {var} = "{file_content}"
+
+            [build.{package_name}]
+            command = """
+                mkdir -p $out/bin
+                cat > $out/{file_path} <<'EOF'
+                    #!/usr/bin/env bash
+                    echo "${var}"
+            EOF
```

Comment:
(non-blocking) Indentation looks weird here and causes leading whitespace in the script but it doesn't affect the shebang or bash and it's partly a result of how we treat indented strings in the manifest 😜

Reply:
Is there an alternative you'd prefer?

Reply:
De-indent everything by 4 spaces like we have done for hooks in floxenvs? Non-blocking though.

Reply:
Done